synapse/api/constants.py

```diff
@@ -73,6 +73,11 @@ class LoginType:
     DUMMY = "m.login.dummy"
 
 
+# This is used in the `type` parameter for /register when called by
+# an appservice to register a new user.
+APP_SERVICE_REGISTRATION_TYPE = "m.login.application_service"
```

Comment:
Why wasn't this added to the `LoginType` class above?

Reply:
See https://github.com/matrix-org/synapse/pull/9548#discussion_r604778903

Reply:
Oh pfft, sorry to have gone around in circles on that issue then.